File: appintro/src/main/java/com/github/paolorotolo/appintro/model/SliderPagerBuilder.java
package com.github.paolorotolo.appintro.model;

import androidx.annotation.ColorInt;
import androidx.annotation.DrawableRes;
import androidx.annotation.FontRes;

public class SliderPagerBuilder {

    private CharSequence title;

    private CharSequence description;

    @DrawableRes
    private int imageDrawable;

    @ColorInt
    private int bgColor;

    @ColorInt
    private int titleColor;

    @ColorInt
    private int descColor;

    private String titleTypeface;

    private String descTypeface;

    @FontRes
    private int titleTypefaceRes;

    @FontRes
    private int descTypefaceRes;

    public SliderPagerBuilder() {

    }

    public SliderPagerBuilder title(CharSequence title) {

        this.title = title;
        return this;
    }

    public SliderPagerBuilder description(CharSequence description) {

        this.description = description;
        return this;
    }

    public SliderPagerBuilder imageDrawable(int imageDrawable) {

        this.imageDrawable = imageDrawable;
        return this;
    }

    public SliderPagerBuilder bgColor(int bgColor) {

        this.bgColor = bgColor;
        return this;
    }

    public SliderPagerBuilder titleColor(int titleColor) {

        this.titleColor = titleColor;
        return this;
    }

    public SliderPagerBuilder descColor(int descColor) {

        this.descColor = descColor;
        return this;
    }

    public SliderPagerBuilder titleTypeface(String titleTypeface) {
        this.titleTypeface = titleTypeface;
        return this;
    }

    public SliderPagerBuilder titleTypefaceRes(@FontRes int titleTypefaceRes) {
        this.titleTypefaceRes = titleTypefaceRes;
        return this;
    }

    public SliderPagerBuilder descTypeface(String descTypeface) {

        this.descTypeface = descTypeface;
        return this;
    }

    public SliderPagerBuilder descTypefaceRes(@FontRes int descTypefaceRes) {
        this.descTypefaceRes = descTypefaceRes;
        return this;
    }

    public SliderPage build() {
        SliderPage sliderPage = new SliderPage();
        sliderPage.setTitle(this.title);
        sliderPage.setDescription(this.description);
        sliderPage.setImageDrawable(this.imageDrawable);
        sliderPage.setBgColor(this.bgColor);
        sliderPage.setTitleColor(this.titleColor);
        sliderPage.setDescColor(this.descColor);
        sliderPage.setTitleTypeface(this.titleTypeface);
        sliderPage.setTitleTypefaceFontRes(this.titleTypefaceRes);
        sliderPage.setDescTypeface(this.descTypeface);
        sliderPage.setDescTypefaceFontRes(this.descTypefaceRes);
        return sliderPage;
    }
}

File: appintro/src/main/java/com/github/paolorotolo/appintro/model/SliderPagerBuilder.kt
package com.github.paolorotolo.appintro.model

import androidx.annotation.ColorInt
import androidx.annotation.DrawableRes
import androidx.annotation.FontRes

/**
 * A builder to help creating [SliderPage] classes.
 * Please use this class only in Java context. From Kotlin just create
 * a [SliderPage] directly.
 */
class SliderPagerBuilder {

    private var title: CharSequence? = null

    private var description: CharSequence? = null

    @DrawableRes
    private var imageDrawable: Int = 0

    @ColorInt
    private var bgColor: Int = 0

    @ColorInt
    private var titleColor: Int = 0

    @ColorInt
    private var descColor: Int = 0

    private var titleTypeface: String? = null

    private var descTypeface: String? = null

    @FontRes
    private var titleTypefaceRes: Int = 0

    @FontRes
    private var descTypefaceRes: Int = 0

    fun title(title: CharSequence): SliderPagerBuilder {
        this.title = title
        return this
    }

    fun description(description: CharSequence): SliderPagerBuilder {
        this.description = description
        return this
    }

    fun imageDrawable(imageDrawable: Int): SliderPagerBuilder {
        this.imageDrawable = imageDrawable
        return this
    }

    fun bgColor(bgColor: Int): SliderPagerBuilder {
        this.bgColor = bgColor
        return this
    }

    fun titleColor(titleColor: Int): SliderPagerBuilder {
        this.titleColor = titleColor
        return this
    }

    fun descColor(descColor: Int): SliderPagerBuilder {
        this.descColor = descColor
        return this
    }

    fun titleTypeface(titleTypeface: String): SliderPagerBuilder {
        this.titleTypeface = titleTypeface
        return this
    }

    fun titleTypefaceRes(@FontRes titleTypefaceRes: Int): SliderPagerBuilder {
        this.titleTypefaceRes = titleTypefaceRes
        return this
    }

    fun descTypeface(descTypeface: String): SliderPagerBuilder {
        this.descTypeface = descTypeface
        return this
    }

    fun descTypefaceRes(@FontRes descTypefaceRes: Int): SliderPagerBuilder {
        this.descTypefaceRes = descTypefaceRes
        return this
    }

    fun build() = SliderPage(
            title = this.title,
            description = this.description,
            imageDrawable = this.imageDrawable,
            bgColor = this.bgColor,
            titleColor = this.titleColor,
            descColor = this.descColor,
            titleTypeface = this.titleTypeface,
            titleTypefaceFontRes = this.titleTypefaceRes,
            descTypeface = this.descTypeface,
            descTypefaceFontRes = this.descTypefaceRes
    )
}
